constraints.go

```diff
@@ -59,6 +61,9 @@ func newConstraints(args ConstraintsArgs) *constraints {
 	copy(tags, args.Tags)
 	spaces := make([]string, len(args.Spaces))
 	copy(spaces, args.Spaces)
+	zones := make([]string, len(args.Zones))
+	copy(zones, args.Zones)
+
 	return &constraints{
 		Version:       1,
```

Comment:
I think we have to bump the version here, because old controllers will otherwise just ignore the Zones being passed, and then it wouldn't provision new machines properly.
We could arguably have "are any zones set, then use V1" sort of logic.
But given the definition has changed, I do think it needs a bump. Maybe I'm missing what the Version means, but I think I have that correct.

Reply:
I looked at some other examples, bumped the version and added a new deserialisation function.